examples/ssh/ssl_certificate.js

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * Domotz Custom Driver 
+ * Name: SSL Certificate Expiration check
+ * Description: This script checks the SSL certificate status for a list of target servers specified in the 'targetServerHost' and 'targetServerPort' parameters.
+ * 
+ * Communication protocol is SSH
+ * 
+ * Tested on Linux Distributions:
+ *      - Ubuntu 20.04 LTS
+ * Shell Version:
+ *      - Bash 5.1.16
+ * 
+ * Return a table containing the following columns for each server:
+ *      - Issuer
+ *      - Expiry
+ *      - Remaining days
+ *      - Is Valid
+ *      - Authorization error
+ * 
+ **/
+
+var table = D.createTable(
+    "SSL Certificates",
+    [
+        { label: "Issuer" },
+        { label: "Expiry" },
+        { label: "Remaining days", unit: "day" },
+        { label: "Is valid" },
+        { label: "Authorization error" }
+    ]
+);
+
+var serversToCheck = D.getParameter('serversToCheck');
+
+var sshConfig = {
+    username: D.device.username(),
+    password: D.device.password(),
+    timeout: 30000
+};
+
+function checkSshError(err) {
+    if(err.message) console.error(err.message);
+    if(err.code == 5) D.failure(D.errorType.AUTHENTICATION_ERROR);
+    if(err.code == 255) D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+    console.error(err);
+    D.failure(D.errorType.GENERIC_ERROR);
+}
+
+function executeCommand(host) {
+    var d = D.q.defer();
+    var command = 'echo | openssl s_client -servername ' + host + ' -connect ' + host + ' 2>/dev/null | openssl x509 -enddate -issuer -noout';
+    sshConfig.command = command;
+    D.device.sendSSHCommand(sshConfig, function (output, error) {
+        if (error) {
+            console.error("Command execution error: " + error);
+            d.reject(error);
+        } else {
+            d.resolve(output);
+        }
+    });
+    return d.promise;
+}
+
+
+function parseValidateOutput(output) {
+    if (output !== "") {
+        console.info("Validation successful");
```

Comment:
Would that work if any of the servers check times out ? Maybe we should allow even if one is OK